variables.tf

```diff
@@ -0,0 +1,39 @@
+variable "prefix" {
+  description = "Prefix for resources created by this module"
+  type        = string
+  default     = "nginx-demo-app"
+}
+
+variable "associate_public_ip_address" {
+  description = "Associate a public IP with the EC2 instance"
+  type        = bool
+  default     = true
+}
+
+variable "vpc_security_group_ids" {
+  description = "List of AWS VPC Security Group IDs"
+  type        = list
+  default     = []
+}
+
+/*variable "vpc_subnet_ids" {
```

Comment:
Kept running into errors with the subnet generation variable so pushing past it for now and will circle back.

Reply:
This may be useful https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/subnet_ids

Additionally, we tackled this by doing the following:

```
# data.tf

data "aws_vpc" "default" {
  default = true
}

data "aws_subnet_ids" "all" {
  vpc_id = data.aws_vpc.default.id
}
```
```
# list all and select the first
subnet_id                   = tolist(data.aws_subnet_ids.all.ids)[0]
```